src/contexts/PermissionsContext.js

```diff
@@ -45,7 +46,42 @@ class PermissionsProvider extends React.Component {
       proxyAddress,
       roleBytes,
     ])
-    console.log('TX', transaction)
+    log('revoke tx:', transaction)
+  }
+
+  // create (set a manager) and grant a permission
+  async createPermission({
+    entityAddress,
+    proxyAddress,
+    roleId = '',
+    roleBytes = null,
+    manager,
+  }) {
+    const { wrapper } = this.props
+    if (wrapper === null) {
+      return
+    }
+    const transaction = await wrapper.performACLIntent('createPermission', [
+      entityAddress,
+      proxyAddress,
+      roleBytes || keccak256(roleId.trim()),
```

Comment:
Interesting, was there a particular case where there weren't `roleBytes` available here?

Reply:
It is to allow people to define a role that is not set, but that is available in the contract. I don’t think we can access the roles of an app if they are not assigned?